### PR TITLE
Truncate response message to avoid too large udp package using UdpMetricsSender

### PR DESCRIPTION
**Pull request by mybreeze77**
In JMeter, we can use "BackendListener - InfluxdbBackendListenerClient - UdpMetricsSender" to send data as udp to InfluxDB. As we know, UDP package max size is limited to the MTU (about 1500 bytes). So it is possible to exceed it, for below example.

```
jmeter,application=application,transaction=all,statut=all count=304,countError=304,hit=304,sb=39216,rb=106688,pct95.0=13.0,pct99.0=16.969999999999985,pct90.0=12.900000000000006 1542959710708000000

jmeter,application=application,statut=all,transaction=HTTP\ Request count=304,avg=8.8,min=6.0,max=17.0,sb=39216,rb=106688,pct95.0=13.0,pct99.0=16.969999999999985,pct90.0=12.900000000000006 1542959710709000000

jmeter,application=application,statut=ko,transaction=HTTP\ Request count=304,avg=8.8,min=6.0,max=17.0,pct95.0=13.0,pct99.0=16.969999999999985,pct90.0=12.900000000000006 1542959710710000000

jmeter,application=application,transaction=HTTP\ Request,responseCode=404,responseMessage=Vvvvvvvvvvvvvvvvvvvvvvvvvvvvvery llllllllllllllllllllllllllllllllllong mmmmmmmmmmmmmmmmmmmmmmessage count=304 1542959710710000000

jmeter,application=application,transaction=internal minAT=1,maxAT=1,meanAT=1,startedT=1,endedT=0 1542959710710000000
```

We will get below error and data are not sent to InfluxDB.

> 2018-09-28 23:53:29,743 ERROR o.a.j.v.b.i.UdpMetricsSender: Error in transferring udp package
> java.io.IOException: Message too long

Considering the size of other fields/tags are under control, it will be better if we truncate the response message to about 50 bytes when it is too large (Only for udp sender).

### CODE REVIEW

File: src/components/org/apache/jmeter/visualizers/backend/influxdb/InfluxdbBackendListenerClient.java [hunk 1]
```diff
@@ -196,12 +196,18 @@ private void addMetrics(String transaction, SamplerMetric metric) {
     }
 
     private void addErrorMetric(String transaction, String responseCode, String responseMessage, long count) {
+        final int MAX_RES_CODE_LENGTH_FOR_UDP = 50;
```
Comment (pmouawad) [inline src/components/org/apache/jmeter/visualizers/backend/influxdb/InfluxdbBackendListenerClient.java]: I think this should be a property.
Can you point me to the reference that mentions 50 ?

Thank you

Comment (mybreeze77) [reply src/components/org/apache/jmeter/visualizers/backend/influxdb/InfluxdbBackendListenerClient.java]: Yes, I agree it is a property which is better to be decided by users. Actually 50 is a kind of empirical value. 

More details said, as we can see below example, "count", "countError", "hit", etc. are integers with 4 bytes. "pct95.0", "pct99.0", etc. are doubles with 8 bytes. But as they will be converted to String in influx line protocal, they actually take about 10 bytes and 20 bytes respectively. Without response message, the data sent to influxdb as udp will be about 1350 bytes at most. 

```
jmeter,application=application,transaction=all,statut=all count=304,countError=304,hit=304,sb=39216,rb=106688,pct95.0=13.0,pct99.0=16.969999999999985,pct90.0=12.900000000000006 1542959710708000000 -- 300 bytes at most

jmeter,application=application,statut=all,transaction=HTTP\ Request count=304,avg=8.8,min=6.0,max=17.0,sb=39216,rb=106688,pct95.0=13.0,pct99.0=16.969999999999985,pct90.0=12.900000000000006 1542959710709000000 -- 300 bytes at most

jmeter,application=application,statut=ko,transaction=HTTP\ Request count=304,avg=8.8,min=6.0,max=17.0,pct95.0=13.0,pct99.0=16.969999999999985,pct90.0=12.900000000000006 1542959710710000000 -- 300 bytes at most

jmeter,application=application,transaction=HTTP\ Request,responseCode=404,responseMessage=Vvvvvvvvvvvvvvvvvvvvvvvvvvvvvery llllllllllllllllllllllllllllllllllong mmmmmmmmmmmmmmmmmmmmmmessage count=304 1542959710710000000 -- 300 bytes at most without response message

jmeter,application=application,transaction=internal minAT=1,maxAT=1,meanAT=1,startedT=1,endedT=0 1542959710710000000 -- 150 bytes at most
```

So for the size of response message, any value lower than 1500 - 1350 = 150 will be safe. That's the reason we set it 50. Hope it clear.

Comment (FSchumacher) [reply src/components/org/apache/jmeter/visualizers/backend/influxdb/InfluxdbBackendListenerClient.java]: We seem to waste a lot of characters with the number conversion of the percentage values. They seem to be exact to about two digits behind the comma, but are written out to much more. Maybe we could use a better formatting/rounding there, too?

Comment (mybreeze77) [reply src/components/org/apache/jmeter/visualizers/backend/influxdb/InfluxdbBackendListenerClient.java]: I think so. If we can shorten the precision of doubles (like avg, min, max, pct) to 2 or 3 digits, that will help save about 200 bytes. I can provide an implementation if consensus reached. What do you think?

Comment (FSchumacher) [reply src/components/org/apache/jmeter/visualizers/backend/influxdb/InfluxdbBackendListenerClient.java]: The rounding should probably done in a different pull request and it should probably take the magnitude of the number into account. That is the method taken by HdrHistogram as suggested to use in #380.

**Comment by mybreeze77** [comment]
In this pr, make “responseMessageSize” as a property (0 means no limitation). Please kindly modify any description if needs. Thanks.

**Comment by mybreeze77** [comment]
Hey, I think we miss something important. The data sent to InfluxDB once also depend on the category of requests. That is, for example, if we define lots of different HTTP requests in one thread group and most of them are failed, the size of data sent to InfluxDB will increase a lot, because each failure will be a "line protocol".

Obviously it doesn't make sense restricting the category of requests, I think this PR also doesn't help resolve the problem. Sorry for not thinking comprehensively. Maybe we can find another way out.

**Comment by vlsi** [comment]
Note: this PR was automatically closed when JMeter development moved from SVN to Git (main branch was renamed from `trunk` to `master`, thus GitHub closed the PR).
If you want to revive the PR, please update your fork as per https://github.com/vlsi/jmeter-git-cleanup/tree/v2.2.0/clean_fork , reopen the PR (unfortunately, it has to be a new PR).

If you have questions, please feel free to ask on JMeter dev list.

Sorry for the inconvenience

**Pull request closed without merging**